### ISSUE TITLE
Did None handling change in Fable?

### ISSUE THREAD

**Issue by halcwb** [fable-compiler/Fable#3218]
### Description

I get a problem that mui is complaining of an undefined value setting. So, the value of the below autocomplete should be set to either a object or a null value. If this is not done on the first render the element is considered to be 'uncontrolled'. 

In the below code I try to set the value, assuming that `None` will be represented as `null`. This worked previously but now seems to be broken:

> Material-UI: A component is changing the uncontrolled value state of Autocomplete to be controlled.
> Elements should not switch from uncontrolled to controlled (or vice versa).
> Decide between using a controlled or uncontrolled Autocomplete element for the lifetime of the component.
> The nature of the state is determined during the first render, it's considered controlled if the value is not `undefined`.
> 

### Repro code

```fsharp
    // It seems that options need to be an array of objects.
    // Make the objects optional to allow for the case of no
    // selected object
    let mapOptions sl =
        sl
        |> List.map (fun s -> {| label = s |} |> Some)
        |> List.toArray
        |> Array.sort


    let applyToOption def f (o: {| label: string |} option) =
        match o with
        | Some o -> o.label |> f
        | None -> def()


    [<ReactComponent>]
    let View (props : Props) =
        let classes = useStyles ()

        let state, dispatch = React.useElmish (init props.Selected, update props.Dispatch, [| box props.Selected; box props.Options |])
        let opts = props.Options |> mapOptions

        Mui.autocomplete
            [ 
                if ((opts |> Array.length) = 1) && props.SetValueIfOne then
                    autocomplete.value (opts |> Array.head)
                else 
                    state.Selected
                    |> Option.map (fun s -> {| label = s |})
                    |> autocomplete.value 

                autocomplete.options opts

                autocomplete.getOptionLabel (applyToOption (fun _ ->  "") id)
                autocomplete.autoComplete true
                autocomplete.renderInput (fun pars ->
                    Mui.textField [ 
                        yield! pars.felizProps
                        textField.label props.Label 
                    ]
                )

                autocomplete.onChange (applyToOption (fun _ -> Clear |> dispatch) (Select >> dispatch))
                autocomplete.getOptionSelected (=)

                autocomplete.filterOptions (fun (options: {| label: string |} option []) (state: string) ->
                    options
                    |> Array.filter (fun o ->
                        match o with
                        | Some o ->
                            let label, state =
                                match props.Filter with
                                | Filter.StartsWith
                                | Filter.Contains -> o.label.Trim().ToLower(), state.Trim().ToLower()
                                | _ -> o.label.Trim(), state.Trim()
                            match props.Filter with
                            | Filter.StartsWith
                            | Filter.StartsWithCaseSensitive -> label.StartsWith(state)
                            | Filter.Contains
                            | Filter.ContainsCaseSensitive -> label.Contains(state)
                            | Filter.Exact -> label = state
                        | None -> false)) 
                ]

```

### Expected and actual results

I would expected no mui error error message as I do set the value at first render? Also, the setting of a value when there is only one value doesn't work. So, it seems a problem with setting values with above code.

Any help greatly appreciated!

### Related information

* Fable version: 3.7.18
* Operating system

**Comment by halcwb** [comment]
This code fixes the above problem:

```fsharp
                if ((opts |> Array.length) = 1) && props.SetValueIfOne then
                    prop.custom ("value", opts |> Array.head |> box)
                    //autocomplete.value (opts |> Array.head)
                else 
                    state.Selected
                    |> Option.map (fun s -> {| label = s |})
                    |> function
                    | None -> prop.custom ("value", null)
                    | Some v -> prop.custom ("value", box v)

```

**Comment by alfonsogarciacaro** [comment]
We did change the `None` representation to `undefined` to fix an issue with optional arguments (see #2032) but it happened two years ago already.

If you can use the work around that'd be great. You can also use `Option.toNullable` although `Nullable` is restricted to value types. Another alternative could be this helper:

```fsharp
[<Emit("$0 ?? null")>]
let asNullable (x: 'T option): 'T option = nativeOnly
```

Fable always use non-strict equality for options (`== null`) which in JS works both for null and undefined, so it's ok to have to handle these options later in your own code.

**Comment by halcwb** [comment]
@alfonsogarciacaro Thanks! I think the original code was indeed written about 2 years ago.
